### PR TITLE
build artifacts

### PR DESCRIPTION
**Pull request by TT-billteng**

### CODE REVIEW

**Review by eyonland — APPROVED** [review]

File: .github/workflows/run-profiler-regression.yaml [hunk 1]
```diff
@@ -1,8 +1,7 @@
 name: "[post-commit] metal - Run profiler regression"
 
 on:
-  push:
```
[the remark this answers is not in the document]
Comment (TT-billteng) [reply .github/workflows/run-profiler-regression.yaml:4]: yes, @ttmchiou is merge multi-queue into fast dispatch tests

profiler build is only built and used in one place so I didn't think we needed an artifact for it

File: .github/workflows/eager-package-main.yaml [hunk 2]
```diff
@@ -4,8 +4,6 @@ name: "[post-commit] eager - Package on main"
 on:
   workflow_dispatch:
   workflow_call:
-  push:
-    branches: ["main"]
```
[the remark this answers is not in the document]
Comment (TT-billteng) [reply .github/workflows/eager-package-main.yaml:8]: this gets triggered when all-post-commit workflow gets triggered on a main push, no?

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -1,8 +1,6 @@
 name: "[post-commit] Build C++ binaries with all configs"
 
 on:
-  push:
-    branches: ["main"]
```
[the remark this answers is not in the document]
Comment (TT-billteng) [reply .github/workflows/build.yaml:5]: this gets triggered when all-post-commit workflow gets triggered on a main push, no?

File: .github/workflows/build-artifact.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+on:
+  workflow_call:
+  workflow_dispatch:
+
+jobs:
+  build-artifact:
+    strategy:
+      # Do not fail-fast because we need to ensure all tests go to completion
+      # so we try not to get hanging machines
+      fail-fast: false
```
[the remark this answers is not in the document]
Comment (TT-billteng) [reply .github/workflows/build-artifact.yaml:12]: k will remove

File: .github/workflows/build-and-unit-tests.yaml [hunk 5]
```diff
@@ -1,9 +1,6 @@
-name: "[post-commit] Slow Dispatch unit tests"
+name: "[post-commit] Slow Dispatch unit tests impl"
```
[the remark this answers is not in the document]
Comment (TT-billteng) [reply .github/workflows/build-and-unit-tests.yaml:1]: what's a good name? I'd prefer if it didn't show up in the list to begin with :|

File: .github/workflows/all-static-checks.yaml [hunk 6]
```diff
@@ -2,8 +2,6 @@ name: "[post-commit] all - Static checks, linters etc."
 
 on:
   workflow_dispatch:
-  push:
-    branches: ["main"]
```
[the remark this answers is not in the document]
Comment (TT-billteng) [reply .github/workflows/all-static-checks.yaml:6]: this gets triggered when all-post-commit workflow gets triggered on a main push, no?

File: .github/workflows/build-and-unit-tests.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (tt-rkim) [reply .github/workflows/build-and-unit-tests.yaml:1]: @ttmchiou Ideas?
Maybe `[internal] Slow Dispatch unit tests impl`

[the remark this answers is not in the document]
Comment (TT-billteng) [reply .github/workflows/build-and-unit-tests.yaml:1]: ok I like

[the remark this answers is not in the document]
Comment (TT-billteng) [reply .github/workflows/build-and-unit-tests.yaml]: updated

**Review by tt-rkim — APPROVED** [review]

**Pull request merged**